src/Catalyst.Cli.Tests/IntegrationTests/Commands/CliCommandTestBase.cs

```diff
@@ -0,0 +1,127 @@
+    {
+        private protected static readonly string ServerNodeName = "node1";
+        private protected static readonly string NodeArgumentPrefix = "-n";
+        protected IRpcClient RpcClient;
+        protected ILifetimeScope Scope;
+        protected ICatalystCli Shell;
+
+        protected CliCommandTestsBase(ITestOutputHelper output) : base(output, new[]
+        {
+            Path.Combine(Constants.ConfigSubFolder, TestConstants.TestShellNodesConfigFile),
+            Path.Combine(Constants.ConfigSubFolder, CliConstants.ShellConfigFile),
+        })
+        {
+            ConfigureModules();
+
+            ConfigureNodeClient();
+
+            CreateResolutionScope();
+
+            ConnectShell();
+        }
+
+        private void ConfigureModules()
+        {
+            var containerBuilder = ContainerProvider.ContainerBuilder;
+            CatalystCliBase.RegisterClientDependencies(containerBuilder);
+        }
+
+        private void CreateResolutionScope()
+        {
+            Scope = ContainerProvider.Container.BeginLifetimeScope(CurrentTestName);
+        }
+
+        private void ConnectShell()
+        {
+            Shell = Scope.Resolve<ICatalystCli>();
+            var hasConnected = Shell.ParseCommand("connect", NodeArgumentPrefix, ServerNodeName);
+            hasConnected.Should().BeTrue();
+        }
+
+        protected void ConfigureNodeClient()
+        {
+            var channel = Substitute.For<IChannel>();
+            channel.Active.Returns(true);
+
+            RpcClient = Substitute.For<IRpcClient>();
+            RpcClient.Channel.Returns(channel);
+            RpcClient.Channel.RemoteAddress.Returns(new IPEndPoint(IPAddress.Loopback, IPEndPoint.MaxPort));
+
+            var nodeRpcClientFactory = Substitute.For<IRpcClientFactory>();
+            nodeRpcClientFactory
+               .GetClient(Arg.Any<X509Certificate2>(), Arg.Is<IRpcClientConfig>(c => c.NodeId == ServerNodeName))
+               .Returns(RpcClient);
+
+            ContainerProvider.ContainerBuilder.RegisterInstance(nodeRpcClientFactory).As<IRpcClientFactory>();
+        }
+
+        protected T AssertSentMessageAndGetMessageContent<T>() where T : IMessage<T>
+        {
+            RpcClient.Received(1).SendMessage(Arg.Is<IMessageDto<ProtocolMessage>>(x =>
+                x.Content != null &&
+                x.Content.GetType().IsAssignableTo<ProtocolMessage>() &&
+                x.Content.FromProtocolMessage<T>() != null
+            ));
+            var sentMessageDto = (IMessageDto<ProtocolMessage>) RpcClient.ReceivedCalls()
+               .Single(c => c.GetMethodInfo().Name == nameof(IRpcClient.SendMessage))
+               .GetArguments()[0];
+            var requestSent = sentMessageDto.Content.FromProtocolMessage<T>();
+            return requestSent;
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing) Scope?.Dispose();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)